packages/event-handler/tests/unit/rest/BaseRouter.test.ts

```diff
@@ -0,0 +1,158 @@
+import context from '@aws-lambda-powertools/testing-utils/context';
+import type { Context } from 'aws-lambda';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { BaseRouter } from '../../../src/rest/BaseRouter.js';
+import type { ResolveOptions } from '../../../src/types/index.js';
+import type {
+  RouteHandler,
+  RouteOptions,
+  RouterOptions,
+} from '../../../src/types/rest.js';
+
+describe('BaseRouter', () => {
+  class TestResolver extends BaseRouter {
+    public readonly handlers: Map<string, RouteHandler> = new Map();
+
+    constructor(options?: RouterOptions) {
+      super(options);
+      this.logger.debug('test debug');
+      this.logger.warn('test warn');
+      this.logger.error('test error');
+    }
+
+    #isEvent(obj: unknown): asserts obj is { path: string; method: string } {
+      if (
+        typeof obj !== 'object' ||
+        obj === null ||
+        !('path' in obj) ||
+        !('method' in obj)
+      ) {
+        throw new Error('Invalid event object');
+      }
+    }
+
+    public route(handler: RouteHandler, options: RouteOptions) {
+      if (options.path == null || options.method == null)
+        throw new Error('path or method cannot be null');
+      this.handlers.set(options.path + options.method, handler);
+    }
+
+    public resolve(
+      event: unknown,
+      context: Context,
+      options?: ResolveOptions
+    ): Promise<unknown> {
+      this.#isEvent(event);
+      const { method, path } = event;
+      const handler = this.handlers.get(path + method);
+      if (handler == null) throw new Error('404');
+      return handler(event, context);
+    }
+  }
+
+  beforeEach(() => {
+    vi.unstubAllEnvs();
+  });
+
+  it.each([
+    ['GET', 'get'],
+    ['POST', 'post'],
+    ['PUT', 'put'],
+    ['PATCH', 'patch'],
+    ['DELETE', 'delete'],
+    ['HEAD', 'head'],
+  ])('should route %s requests', async (method, verb) => {
+    const app = new TestResolver();
+    (
+      app[
+        verb as 'get' | 'post' | 'put' | 'patch' | 'delete' | 'head'
```

Comment:
Should we use the `HttpVerb` constant values here?

Reply:
Done